### PR TITLE
adding isFetchedFlag, no cache time to force API calls, disabled retries

### PR DESCRIPTION
**Pull request by rsimosa**
- added a query cache time of 0 to force post on all downloads instead of react trying to use the cache
- added retry false to react useQuery hook as it was going into endless loop trying to get the files
- updated logic for download button to only show if its logged in, is not error, and is not currently downloading
- added readOnly attribute for some query's 
- added distinctBy given Id's

### CODE REVIEW

File: src/API/WesternStatesWater.WestDaat.Accessors/WaterAllocationAccessor.cs [hunk 1]
```diff
@@ -282,13 +285,13 @@ IEnumerable<IEnumerable<object>> IWaterAllocationAccessor.GetWaterRights(WaterRi
 
             var response = new List<IEnumerable<object>>()
             {
-                waterRightDetails.Select(x=>x.VariableSpecific).ProjectTo<CsvModels.Variables>(DtoMapper.Configuration),
-                waterRightDetails.Select(x=>x.Organization).ProjectTo<CsvModels.Organizations>(DtoMapper.Configuration),
-                waterRightDetails.Select(x=>x.Method).ProjectTo<CsvModels.Methods>(DtoMapper.Configuration),
-                filteredSites.SelectMany(x=>x.AllocationBridgeSitesFact).ProjectTo<CsvModels.WaterAllocations>(DtoMapper.Configuration),
+                waterRightDetails.Select(x=>x.VariableSpecific).ProjectTo<CsvModels.Variables>(DtoMapper.Configuration).AsEnumerable().DistinctBy(x=>x.VariableSpecificUuid),
```
Comment (EverettRhodes) [inline src/API/WesternStatesWater.WestDaat.Accessors/WaterAllocationAccessor.cs:288]: What is the reason for calling `.AsEnumerable()` before the `.DistinctBy`? Calling `.AsEnumerable()` executes the query and loads the results into memory in the function app before applying the `DistinctBy`

File: src/API/WesternStatesWater.WestDaat.Managers/WaterAllocationManager.cs [hunk 2]
```diff
@@ -204,7 +204,7 @@ public void WaterRightsAsZip(Stream responseStream, ClientContracts.WaterRightsS
                         csv.Context.TypeConverterOptionsCache.GetOptions<DateTime>().Formats = new string[] { "d" };
                         csv.Context.TypeConverterOptionsCache.GetOptions<DateTime?>().Formats = new string[] { "d" };
 
-                        csv.WriteRecords(file);
+                        csv.WriteRecords(file.Distinct());
```
Comment (EverettRhodes) [inline src/API/WesternStatesWater.WestDaat.Managers/WaterAllocationManager.cs:207]: What does calling `Distinct()` here provide?

**Review by EverettRhodes — CHANGES_REQUESTED** [review]

File: src/API/WesternStatesWater.WestDaat.Managers/WaterAllocationManager.cs (hunk 2 above)
Comment (rsimosa) [reply src/API/WesternStatesWater.WestDaat.Managers/WaterAllocationManager.cs:207]: Nothing.. I missed it when I was trying different approaches

File: src/API/WesternStatesWater.WestDaat.Accessors/WaterAllocationAccessor.cs (hunk 1 above)
Comment (rsimosa) [reply src/API/WesternStatesWater.WestDaat.Accessors/WaterAllocationAccessor.cs:288]: to be able to do a DistinctBy it needs to be an Enumerable, still, it won't run the queries until its writing into the file keeping the memory usage low

**Review by EverettRhodes — APPROVED** [review]

**Pull request merged**